OfflineService/OfflineServiceViewController.swift

```diff
@@ -0,0 +1,210 @@
+    }
+
+    // MARK: - Setting-up methods
+    
+    private func setupUI() {
+        tableView.register(UINib(nibName: OfflineDataRegionTableViewCell.identifier, bundle: nil),
+                           forCellReuseIdentifier: OfflineDataRegionTableViewCell.identifier)
+        
+        tableView.separatorInset = .zero
+        tableView.allowsSelection = true
+        
+        title = OfflineServiceConstants.title
+        
+        navigationItem.leftBarButtonItem = UIBarButtonItem(image: UIImage(named: "settings"),
+                                                           style: .plain,
+                                                           target: self,
+                                                           action: #selector(showSettings))
+        
+        navigationItem.rightBarButtonItem = UIBarButtonItem(title: OfflineServiceConstants.close,
+                                                            style: .done,
+                                                            target: self,
+                                                            action: #selector(dismissViewController))
+    }
+    
+    private func setupOfflineServiceDataSource() {
+        offlineServiceDataSource = OfflineServiceDataSource()
+        offlineServiceDataSource?.delegate = self
+        offlineServiceDataSource?.startObservingAvailableRegions()
+    }
+    
+    // MARK: - Action handler methods
+    
+    @objc private func dismissViewController() {
+        dismiss(animated: true, completion: nil)
+    }
+    
+    @objc private func showSettings() {
+        let alertController = UIAlertController(title: OfflineServiceConstants.title,
+                                                message: OfflineServiceConstants.selectActionTitle,
+                                                preferredStyle: .alert)
+        
+        let clearMapsAmbientCacheActionHandler: ActionHandler = { _ in
+            clearMapsAmbientCache()
+        }
+        
+        let clearNavigationAmbientCacheActionHandler: ActionHandler = { _ in
+            clearNavigationAmbientCache()
+        }
+        
+        let actionPayloads: [(String, UIAlertAction.Style, ActionHandler?)] = [
+            (OfflineServiceConstants.clearMapsAmbientCache, .default, clearMapsAmbientCacheActionHandler),
+            (OfflineServiceConstants.clearNavigationAmbientCache, .default, clearNavigationAmbientCacheActionHandler),
+            (OfflineServiceConstants.cancel, .cancel, nil)
+        ]
+        
+        actionPayloads
+            .map({ payload in UIAlertAction(title: payload.0, style: payload.1, handler: payload.2) })
+            .forEach(alertController.addAction(_:))
+        
+        present(alertController, animated: true, completion: nil)
+    }
+    
+    // MARK: - OfflineServiceDataSourceDelegate methods
+    
+    func offlineServiceDataSource(_ dataSource: OfflineServiceDataSource, didUpdate offlineDataItems: [OfflineDataItem]) {
```

Comment:
there might be a situation when the following call of this method will contain a sublist of regions from the previous call, in that case we should remove some items from the `offlineDataItems`. I understand that in the current usage it's not a problem because the view controller most likely will be recreated between the calls, but there is no guarantees that in the future changes there will no be such situations